src/poetry/installation/executor.py

```diff
@@ -909,3 +923,40 @@ def _get_archive_info(self, package: Package) -> dict[str, Any]:
             archive_info["hashes"] = {algorithm: value}
 
         return archive_info
+
+    @staticmethod
+    def _get_cached_archive_for_link(
+        env: Env, cache_dir: Path, link: Link, *, strict: bool
+    ) -> Path | None:
```

Comment:
Thinking a bit further, in the follow up PR, we will probably want to call this method (with `strict=True`) in the `Provider`. Thus, it might make sense to move this method to `cache.py`, too. Further, `env` is only required if `strict=False`. I think we should make `env` a keyword-only parameter with default `None` and add an `assert strict or env is not None` at the beginning of the method.

Reply:
I agree in general, and that's how I first refactored it. But since `get_cached_archive_for_link` depends from classes in `installation` (`Wheel` and `InvalidWheelName`) I ended up in a circular import issue that could be solved only by emptying `poetry/installation/__init__.py`. At that point though other things, such as the `poetry-export-plugin`, started failing... While I know that it's not the main codebase's responsibility to make sure that plugins don't break, I felt that it would be better avoiding possibly breaking plugins if there isn't a really strong reason for doing that.

I propose you a couple alternative solutions resulting `get_cached_archive_for_link` in `utils.cache.py`, let me know which one you prefer:
1. `installation` is not touched, we `import poetry.installation.choose.Wheel` _inside_ the function to avoid circular import
1. `poetry.installation.chooser.Wheel` and `poetry.installation.chooser.InvalidWheelName` are moved into a separate utils file `poetry.utils.wheel`
1. Duplicate the small wheel logic we need, thus making `cache.py` and `Wheel` completely decoupled
```python
# NOW
from poetry.installation.chooser import InvalidWheelName
from poetry.installation.chooser import Wheel

try:
    wheel = Wheel(archive.name)
except InvalidWheelName:
    continue
```

Reply:
A fourth alternative would be to split `get_cached_archive_for_link` into a strict version that is moved to `utils.cache` and a non-strict version that is kept in `executor`.

However, I think your variant 2 sounds good.

Reply:
I was just performing the change to variant 2 :D Now I guess I'm just missing the class-based refactor. I'll ask you for another review once I closed all comments. Thanks a lot!